Lib/pathlib.py

```diff
@@ -529,23 +529,23 @@ def _select_from(self, parent_path, is_dir, exists, scandir):
         try:
             entries = list(scandir(parent_path))
             for entry in entries:
-                entry_is_dir = False
-                try:
-                    entry_is_dir = entry.is_dir()
-                except OSError as e:
-                    if not _ignore_error(e):
-                        raise
-                if not self.dironly or entry_is_dir:
-                    name = entry.name
-                    if self.match(name):
-                        path = parent_path._make_child_relpath(name)
-                        for p in self.successor._select_from(path, is_dir, exists, scandir):
-                            yield p
+                if self.dironly:
+                    try:
+                        if not entry.is_dir():
```

Comment:
Perhaps add a comment mentioning that it can raise a `PermissionError` which is not part of the ignored errnos?

Reply:
Done in 38894